src/common/types/varint.cpp

```diff
@@ -34,7 +34,7 @@ void Varint::Verify(const string_t &input) {
 	//  No bytes between 4 and end can be 0, unless total size == 4
 	if (varint_bytes > 4) {
 		if (is_negative) {
-			if (~varint_ptr[3] == 0) {
+			if (static_cast<data_t>(~varint_ptr[3]) == 0) {
```

Comment:
I think the sanity check here is to make sure not all bits are 1 for `varint_ptr[3]`; let me know if I mis-understand.